modules/swagger-core/src/main/java/io/swagger/core/filter/SpecFilter.java

```diff
@@ -66,17 +66,19 @@ public Swagger filter(Swagger swagger, SwaggerSpecFilter filter, Map<String, Lis
         for (String key : definitions.keySet()) {
             Model definition = definitions.get(key);
             Map<String, Property> clonedProperties = new LinkedHashMap<String, Property>();
-            for (String propName : definition.getProperties().keySet()) {
-                Property property = definition.getProperties().get(propName);
-                if (property != null) {
-                    boolean shouldInclude = filter.isPropertyAllowed(definition, property, propName, params, cookies, headers);
-                    if (shouldInclude) {
-                        clonedProperties.put(propName, property);
+            if (definition.getProperties() != null) {
+                for (String propName : definition.getProperties().keySet()) {
+                    Property property = definition.getProperties().get(propName);
+                    if (property != null) {
+                        boolean shouldInclude = filter.isPropertyAllowed(definition, property, propName, params, cookies, headers);
+                        if (shouldInclude) {
+                            clonedProperties.put(propName, property);
+                        }
                     }
                 }
             }
             Model clonedModel = (Model) definition.clone();
-            clonedModel.getProperties().clear();
+            if (clonedModel.getProperties() != null) clonedModel.getProperties().clear();
```

Comment:
Based on our [coding guidelines](https://github.com/swagger-api/swagger-core/wiki/Coding-Standards#block-less-loops-and-if-statements) can you turn the `if`statement to a blocked one?